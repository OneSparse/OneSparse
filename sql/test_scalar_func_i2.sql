-- # Scalar `smallint` Functions
--
-- This documentation is also tests for the code, the examples below
-- show the literal output of these statements from Postgres.
--
-- Some setup to make sure warnings are shown, and that the extension
-- is installed.
set client_min_messages = 'WARNING';
create extension if not exists onesparse;

-- Test construction of min, zero and max values:
select '-1'::smallint::scalar;
select '0'::smallint::scalar;
select '1'::smallint::scalar;

-- Test setting a scalar value from max to 2
select set('1'::smallint::scalar, 2);

-- Test various casting functions used by the CREATE CAST machinery:
select scalar_smallint((-1)::smallint);
select scalar_smallint((0)::smallint);
select scalar_smallint((1)::smallint);
select smallint_scalar((-1)::smallint::scalar);
select smallint_scalar((0)::smallint::scalar);
select smallint_scalar((1)::smallint::scalar);
