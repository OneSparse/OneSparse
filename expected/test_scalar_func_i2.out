-- # Scalar Functions
--
-- This documentation is also tests for the code, the examples below
-- show the literal output of these statements from Postgres.
--
-- Some setup to make sure warnings are shown, and that the extension
-- is installed.
set client_min_messages = 'WARNING';
create extension if not exists onesparse;
-- Test construction of min, zero and max values:
select '-1'::smallint::scalar;
 scalar 
--------
 i2:-1
(1 row)

select '0'::smallint::scalar;
 scalar 
--------
 i2:0
(1 row)

select '1'::smallint::scalar;
 scalar 
--------
 i2:1
(1 row)

-- Test setting a scalar value from max to 2
select set('1'::smallint::scalar, 2);
 set  
------
 i2:2
(1 row)

-- Test various casting functions used by the CREATE CAST machinery:
select scalar_smallint((-1)::smallint);
 scalar_smallint 
-----------------
 i2:-1
(1 row)

select scalar_smallint((0)::smallint);
 scalar_smallint 
-----------------
 i2:0
(1 row)

select scalar_smallint((1)::smallint);
 scalar_smallint 
-----------------
 i2:1
(1 row)

select smallint_scalar((-1)::smallint::scalar);
 smallint_scalar 
-----------------
              -1
(1 row)

select smallint_scalar((0)::smallint::scalar);
 smallint_scalar 
-----------------
               0
(1 row)

select smallint_scalar((1)::smallint::scalar);
 smallint_scalar 
-----------------
               1
(1 row)

